### PR TITLE
Totw64

### PR DESCRIPTION
**Pull request by tituswinters**

### CODE REVIEW

File: _includes/side-nav-tips.html [hunk 1]
```diff
@@ -9,6 +9,9 @@
        <a href="{{ site.baseurl }}/tips/55">TotW #55: Name Counting</a></h6>
```
Comment (manshreck) [inline _includes/side-nav-tips.html]: Leave this file out. I'll add it when I change published: true.

File: _posts/2017-10-26-totw-64.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+sidenav: side-nav-tips.html
+published: draft
+permalink: tips/64
+type: markdown
+order: 064
+---
+
+## Change "published:true" to post actual tip
+
+Originally posted as totw/64 on 2013-12-09
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]: I've been using "Dec 9, 2013" format (same for updated date).

File: _posts/2017-10-26-totw-64.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+multi-layer escaping), code clarity drops precipitously.
+
+Luckily, there’s a new C++11 feature that removes this need for escaping: raw
+string literals.
+
+## The Raw String Literal Format
+
+A raw string literal has the following special syntax:
+
+    R"tag(whatever you want to say)tag"
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]: Standardize on:

```cpp
(No indent)Code
```

File: _posts/2017-10-26-totw-64.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+`tag` is a sequence of up to 16 characters (and an empty tag is both OK and
+common). The characters after '"tag(' and before the first following occurrence
+of ')tag"' are used literally as the contents of the string literal. 'tag' may
+contain any character but parentheses, backslash, and whitespace.
+
+Examine the difference:
+
+```
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]:
```cpp

File: _posts/2017-10-26-totw-64.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const char concert_17_raw[] =
+    "id: 17\n"
+    "artist: \"Beyonce\"\n"
+    "date: \"Wed Oct 10 12:39:54 EDT 2012\"\n"
+    "price_usd: 200\n";
+```
+
+versus:
+
+```
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]:
```cpp

File: _posts/2017-10-26-totw-64.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Note that indentation rules, combined with the fact that raw string literals may
+contain newlines, leave you with an awkward choice on how to indent the first
+line of your raw string block. Because text protobufs ignore whitespace, this
+problem can be avoided by throwing in a leading newline (ignored by the parser)
+in that case, but not all uses of raw strings are so forgiving.
+
+A non-empty tag is useful when the sequence `)"` happens to appear in your
+string and therefore can’t act as the closing delimiter:
+
+    std::string my_string = R"foo(This contains quoted parens "()")foo";
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]:
```cpp

File: _posts/2017-10-26-totw-64.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Conclusion
+
+Raw string literals are certainly not an everyday tool for most of us, but there
+are times when good use of this new language feature will increase readability.
+So the next time you’re scratching your head trying to figure out if you need
+two <code>\\</code>s or four, try raw string literals instead. Your readers will
+thank you for it, even if regular expressions are still hard:
+
+    R"regexp((?:"(?:\\"|[^"])*"|'(?:\\'|[^'])*'))regexp";
```
Comment (manshreck) [inline _posts/2017-10-26-totw-64.md]:
```cpp

File: _posts/2017-10-26-totw-64.md (hunk 2 above)
Comment (tituswinters) [reply _posts/2017-10-26-totw-64.md]: The American date format is intolerable. Can we avoid it?

File: _posts/2017-10-26-totw-64.md (hunk 3 above)
Comment (tituswinters) [reply _posts/2017-10-26-totw-64.md]: Done

File: _posts/2017-10-26-totw-64.md (hunk 2 above)
Comment (manshreck) [reply _posts/2017-10-26-totw-64.md]: We can do whatever we want. Certainly the American format is bad for anything simply numeric (and we can't use it for the filename, fwiw), but I do actually prefer "Dec 9, 2013" to "2012-12-09". We should change every date to match the latter then (including here).

**Comment by tituswinters** [comment]
Updated.

**Comment by tituswinters** [comment]
Updated.

**Comment by tituswinters** [comment]
Updated.

**Pull request merged**